### DISCUSSION TITLE
Cannot import dependencies from cuquantum in colab

### DISCUSSION THREAD

**Discussion by Snuag** [NVIDIA/cuQuantum#56] [Q&A] [answered]
When I ran the file :https://github.com/NVIDIA/cuQuantum/blob/main/python/samples/cutensornet/circuit_converter/cirq_advanced.ipynb 
in colab i am getting this error when I import any dependencies of cuquantum
![image](https://github.com/NVIDIA/cuQuantum/assets/100287263/7c5dc4d9-06de-4b6f-837f-7f758fcafcb4)
This is the version of cuquantum installed in colab:
![image](https://github.com/NVIDIA/cuQuantum/assets/100287263/2f98537a-46b1-4897-b8f7-8d0e92e188fb)

**Comment by leofang** [comment]
Converting to a discussion.

**Comment by leofang** [comment]
Hi @Snuag, could you show the output of these lines?
```
!pip show cuquantum-python
!pip show cuquantum-python-cu11
!pip show cuquantum-python-cu12
```

**Comment by Snuag** [comment]
Here is the snippet of the above code:
![image](https://github.com/NVIDIA/cuQuantum/assets/100287263/f03d11ed-7701-4f93-a6ed-e7fe90f3eb9c)
![image](https://github.com/NVIDIA/cuQuantum/assets/100287263/b105379b-4d04-4dc9-a46a-6aa76d52e46f)

**Reply by leofang** [reply]
Hi @Snuag, your package installation may be corrupted. You should not have both `-cu11` and `-cu12` packages installed. How did you set up your Colab session?

I suggest to remove the `-cuXX` package that is not matching your Colab's CUDA version (ex: use `-cu11` for CUDA 11 and so on).

**Comment by Snuag** [comment]
Thank you for quick response @leofang ! I must be doing something wrong with installing packages. I want to run VQE file given in https://github.com/NVIDIA/cuQuantum/blob/main/python/samples/cutensornet/circuit_converter/cirq_advanced.ipynb
Can you please guide me where am I going wrong here? I have attached the colab file:
[cutn.txt](https://github.com/NVIDIA/cuQuantum/files/11481100/cutn.txt)

**Reply by leofang** [reply]
Try replacing all these lines 
```
!pip install cuquantum
!pip install --upgrade cuquantum
!pip install cuquantum-python
!pip install cuquantum-python-cu11
```
with just
```
!pip install cuquantum-python-cu11 cupy-cuda11x
```
or
```
!pip install cuquantum-python-cu12 cupy-cuda12x
```
depending on your CUDA version on Colab (which is unclear to me).

**Reply by Snuag** [reply]
Thank you ! I will update the file and check again!

**Reply by Snuag** [reply]
**CUDA version in colab is 11.8** 
(nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:33:58_PDT_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0)

I followed the steps provided by you but I still face the error:
![image](https://github.com/NVIDIA/cuQuantum/assets/100287263/b6cfbefb-7886-4b38-8dcf-1eb5dd8b4efd)

**Reply by leofang** [reply, marked as answer]
Unfortunately this is more of a Colab question than a cuQuantum one. Any chance you forget to choose/enable GPU as your hardware accelerator on Colab? `libcuda.so` is the DSO for NVIDIA GPU driver, and apparently it's not available to the dynamic loader.

To check if a GPU is detectable & working, we usually recommend running `nvidia-smi`:
```
!nvidia-smi
```

**Reply by Snuag** [reply]
Thank you @leofang ! there was a problem with GPU. It's working now!!!

**Reply by leofang** [reply]
Glad it worked!
